src/Platform/Microsoft.Testing.Platform/Requests/TestHostTestFrameworkInvoker.cs

```diff
@@ -68,6 +70,8 @@ public async Task ExecuteAsync(ITestFramework testFramework, ClientInfo client,
 
     public virtual async Task ExecuteRequestAsync(ITestFramework testFramework, TestExecutionRequest request, IMessageBus messageBus, CancellationToken cancellationToken)
     {
+        await Task.Yield();
```

Comment:
here we ensure that this method will run asynchronously, so we can handle the Task.WhenAny above in that call stack.